radix-engine-interface/src/api/object_api.rs

```diff
@@ -101,6 +101,13 @@ pub trait ClientObjectApi<E> {
     /// Get info regarding a visible object
     fn get_object_info(&mut self, node_id: &NodeId) -> Result<ObjectInfo, E>;
 
+    // TODO: Combine this with globalization process and/or find the right abstraction
```

Comment:
I understand you have a `TODO` here, but I'm a bit concerned with the approach here, as it breaks the "access rules for global objects only" invariant.

It seems to me that we either need something like `auth => require(!vault.is_locked) && [other_constraints]` or we can do `if self.is_locked` check within withdraw methods.

Reply:
I do understand this breaks that invariant but I think special casing for vaults here is the correct approach. We can discuss more as we do architecture refactors.